docs/cluster_management.md

```diff
@@ -2,45 +2,76 @@
-3. Turn off the old version.
+## Configuration options
 
-### Turn on or off an API version for your cluster
+Different environments may have different configurations options, so here we provide only most general and by far not exhaustive configuration options of Kubernetes cluster. Configuration of Kubernetes components (e.g. API server, Kubelet) is passed as commandline flags to the binaries. If environment of your choice is using Salt to setup the cluster you can check how we pass them to the components in `cluster/saltbase/salt` subdirectories.
 
-Specific API versions can be turned on or off by passing --runtime-config=api/<version> flag while bringing up the server. For example: to turn off v1 API, pass --runtime-config=api/v1=false.
-runtime-config also supports 2 special keys: api/all and api/legacy to control all and legacy APIs respectively. For example, for turning off all api versions except v1, pass --runtime-config=api/all=false,api/v1=true.
+Two interesting examples:
+* `max-pods` in Kubelet setting maximal number of Pods that are allowed to run on a single Node,
+* `pod-eviction-timeout` in kube-controller-manager setting how long we wait after Node becomes unresponsive before we start evicting Pods from it.
+More can be seen by running particulars app help.
 
-### Switching your cluster's storage API version
+If you want to change the value of particular flag, e.g. `max-pod`, you need to find a file used by Salt to set Kubelet flags. This particular file is `cluster/saltbase/salt/kubelet/default`. In this file you can modify the line with DAEMON_ARGS by adding the value for the flag you want, e.g. change the DAEMON_ARGS line to
+```
+DAEMON_ARGS="{{daemon_args}} {{api_servers_with_port}} {{debugging_handlers}} {{hostname_override}} {{cloud_provider}} {{config}} --allow_privileged={{pillar['allow_privileged']}} {{pillar['log_level']}} {{cluster_dns}} {{cluster_domain}} {{docker_root}} {{kubelet_root}} {{configure_cbr0}} {{cgroup_root}} {{system_container}} {{pod_cidr}} --max-pods=42"
+```
 
-KUBE_API_VERSIONS env var controls the API versions that are supported in the cluster. The first version in the list is used as the cluster's storage version. Hence, to set a specific version as the storage version, bring it to the front of list of versions in the value of KUBE_API_VERSIONS.
+If you want to have more flexibility for configuring your cluster, you'll need to learn more about the [Salt stack](http://saltstack.com/).
 
-### Switching your config files to a new API version
+## Upgrading a cluster
 
-You can use the kube-version-change utility to convert config files between different API versions.
+GKE handles cluster upgrades for you, so you don't need to worry about. If you already a cluster running in some other environment and want to upgrade Kubernetes version running on it, you can use the `cluster/kube-push.sh` script. It works out of the box if your cluster runs on GCE and most of other providers.
```

Comment:
I think more details about how to use kube-push would be the most useful part of this doc.